scripts/dev/reformat-java

```diff
@@ -1,32 +1,21 @@
 #!/bin/sh
 #
-# @author René Schwaiger <sanssecours@me.com>
+# @author Michael Tucek <michael@tucek.eu>
 # @brief Reformats Java source code
-# @date 10.11.2019
+# @date 30.09.2021
 # @tags reformat
 
-SCRIPTS_DIR=$(dirname "$0")
-. "${SCRIPTS_DIR}/include-common"
-
-REQUIRED_VERSION=12
-CLANG_FORMAT=$(command -v clang-format-$REQUIRED_VERSION || command -v clang-format)
-
-if [ -n "$CLANG_FORMAT" ]; then
-	LOCATION="$CLANG_FORMAT"
-	VERSION=$("$CLANG_FORMAT" --version 2> /dev/null)
-	MAJOR_VERSION=$(printf '%s' "$VERSION" | sed -E 's/.* ([0-9]+)\.[0-9].[0-9][ -]?.*/\1/')
-	if [ "${MAJOR_VERSION:-0}" -ne $REQUIRED_VERSION ]; then
-		unset CLANG_FORMAT
-	fi
+FORMATTER_JAR=/tmp/google-java-format.jar
+FORMATTER_DOWNLOAD_URL=https://github.com/google/google-java-format/releases/download/v1.11.0/google-java-format-1.11.0-all-deps.jar
```

Comment:
I propose the following changes:

1. Place the jar in `/opt` or some directory that is not writable by everyone.
2. Do not download the dependency in the reformat script, but have it either earlier (e.g. by Dockerfile setup) or not at all (as with the rest of the reformat scripts).

Reasoning:

1. We don't want to introduce security vulnerabilities by executing something anyone could have left in `/tmp`. We also want to persist the dependency somewhere.
2. If we download the dependency in this script, it is executed very late. Temporary network failures may cause the tests to fail late due to download problems. It would be much better and simpler if this is setup somewhere else (Dockerfile, cirrus config, github workflow config...).

Do you want / have time to implement the changes before release? If not, please ping me, then I will do it.

Reply:
The debian sid dockerfile already includes the jar in `/tmp/google-java-format.jar`
I thought an automated download would help if anyone executes it locally.
I could include the jar (for the debian sid dockerfile) in `/opt/google-java-format.jar` instead of `/tmp/google-java-format.jar`.
Debian sid is the container mentioned to be used for remormatting in `doc/tutorials/run_reformatting_script_with_docker.md`
To prevent the need for manual download, wehn using the script locally, i could introduce a command line switch for the script?
@mpranj what do you think?

Reply:
> I could include the jar (for the debian sid dockerfile) in /opt/google-java-format.jar instead of /tmp/google-java-format.jar.

That sounds good!

> To prevent the need for manual download, wehn using the script locally, i could introduce a command line switch for the script?

That's optional to me, but if you want to provide it go ahead 👍

Reply:
done